app/components/widgets/forms/location-input.js

```diff
@@ -24,7 +25,10 @@ export default Component.extend({
 
   actions: {
     showAddressView(show = true) {
-      this.set('addressViewIsShown', show);
+      let showAddress = this.get('showButton');
```

Comment:
When the user pressed Return key the callback `placeChanged` is triggered which then triggers the `showAddressView` which hides the input field and shows the location in address form.